lldb/source/Plugins/ObjectFile/Mach-O/ObjectFileMachO.cpp

```diff
@@ -6621,29 +6624,28 @@ bool ObjectFileMachO::SaveCore(const lldb::ProcessSP &process_sp,
           LC_THREAD_data.SetAddressByteSize(addr_byte_size);
           LC_THREAD_data.SetByteOrder(byte_order);
         }
-        for (uint32_t thread_idx = 0; thread_idx < num_threads; ++thread_idx) {
-          ThreadSP thread_sp(thread_list.GetThreadAtIndex(thread_idx));
```

Comment:
we need to revert this change back as we need a zero based thread index into `LC_THREAD_datas`:
```
for (uint32_t thread_idx = 0; thread_idx < num_threads; ++thread_idx) {
  ThreadSP thread_sp = thread_list[thread_idx];
```
We don't want to use  `for (const ThreadSP &thread_sp : thread_list) {` anymore because we need the index for the `thread_sp` within `thread_list`